sentry_sdk/integrations/graphene.py

```diff
@@ -106,3 +114,42 @@ def _event_processor(event, hint):
         del event["request"]["data"]
 
     return event
+
+
+@contextmanager
+def graphql_span(schema, source, kwargs):
+    # type: (GraphQLSchema, Union[str, Source], Dict[str, Any]) -> Generator[None, None, None]
+    operation_name = kwargs.get("operation_name")
+
+    operation_type = "query"
+    op = OP.GRAPHQL_QUERY
+    if source.strip().startswith("mutation"):
+        operation_type = "mutation"
+        op = OP.GRAPHQL_MUTATION
+    elif source.strip().startswith("subscription"):
+        operation_type = "subscription"
+        op = OP.GRAPHQL_SUBSCRIPTION
+
+    sentry_sdk.add_breadcrumb(
+        crumb={
+            "data": {
+                "operation_name": operation_name,
+                "operation_type": operation_type,
+            },
+            "category": "graphql.operation",
+        },
+    )
+
+    scope = Scope.get_current_scope()
+    if scope.span:
+        _graphql_span = scope.span.start_child(op=op, description=operation_name)
+    else:
+        _graphql_span = sentry_sdk.start_span(op=op, description=operation_name)
+
+    _graphql_span.set_data("graphql.document", source)
+    _graphql_span.set_data("graphql.operation.name", operation_name)
+    _graphql_span.set_data("graphql.operation.type", operation_type)
+
+    yield
+
+    _graphql_span.finish()
```

Comment:
We probably need to make sure we finish the span even in the case that we have an error in the `yield`
```suggestion
    try:
        yield
    finally:
         _graphql_span.finish()
```